File: crack.c
#define _XOPEN_SOURCE
#include <unistd.h>
#include <stdlib.h>
#include <stdio.h>
#include <cs50.h>
#include <crypt.h>
#include <string.h>

int main (int argc, char *argv[])
{
	if (argc != 2)                      //checking for existence of 2 arguments
	{
		printf("Usage: ./crack hash\n");
		return 1;
	}
	char *crypt_1 = argv[1] ;
	char str[5];
    int rez = 1;
    //Cracking passwords from 1 to 4 characters of it
    for(int i = 2; rez != 0; i++)         //promt for array size
    {
        for(int m = 65; m < 122; m++)
        {
            str[i-1] = '\0';
            str[i-2] = m;               //for password consisting of 1 characters
            if (strcmp(crypt(str, "50"), crypt_1) == 0)
            {
                rez = 0;
                printf("%s\n", str); //PASSWORD
                break;
            }
            else if (i > 2)             //for password consisting of 2 characters
            {
                for(int l = 65; l < 122; l++)
                {
                    str[i-3] = l;
                    if (strcmp(crypt(str, "50"), crypt_1) == 0)
                    {
                        rez = 0;
                        printf("%s\n", str); //PASSWORD
                        break;
                    }
                    else if (i > 3)         //for password consisting of 3 characters
                    {
                        for(int k = 65; k < 122; k++)
                        {
                            str[i-4] = k;
                            if (strcmp(crypt(str, "50"), crypt_1) == 0)
                            {
                                rez = 0;
                                printf("%s\n", str); //PASSWORD
                                break;
                            }
                            else if (i > 4)         //for password consisting of 4 characters
                            {
                                for(int j = 65; j < 122; j++)
                                {
                                    str[i-5] = j;
                                    if (strcmp(crypt(str, "50"), crypt_1) == 0)
                                    {
                                        rez = 0;
                                        printf("%s\n", str); //PASSWORD
                                        break;
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
return 0;
}

File: crack.py
import sys
import os
import itertools as it
import crypt

if len(sys.argv) is not 2:    # checking for existence of 2 arguments
    print("Usage: python crack.py hash")
    exit(1)

crypt_1 = sys.argv[1]
str1 = []
rez = 1
# Cracking passwords from 1 to 4 characters of it
for i in range(1, 5):  # promt for array size
    str1.append("")
    for m in it.chain(range(65, 91), range(97, 123)):
        str1[0] = (chr(m))    # for password consisting of 1 characters
        str2 = ''.join(map(str, str1))
        if (crypt.crypt(str2, '50') == crypt_1):
            rez = 0
            print("{}" .format(str2))  # PASSWORD
            exit(0)
        elif (i > 1):  # for password consisting of 2 characters
            for l in it.chain(range(65, 91), range(97, 123)):
                str1[1] = (chr(l))
                str2 = ''.join(map(str, str1))
                if (crypt.crypt(str2, '50') == crypt_1):
                    rez = 0
                    print("" .join(str2))  # PASSWORD
                    exit(0)
                elif (i > 2):         # for password consisting of 3 characters
                    for k in it.chain(range(65, 91), range(97, 123)):
                        str1[2] = (chr(k))
                        str2 = ''.join(map(str, str1))
                        if (crypt.crypt(str2, '50') == crypt_1):
                            rez = 0
                            print("" .join(str2))  # PASSWORD
                            exit(0)
                        elif (i > 3):  # for password consisting of 4 characters
                            for j in it.chain(range(65, 91), range(97, 123)):
                                str1[3] = (chr(j))
                                str2 = ''.join(map(str, str1))
                                if (crypt.crypt(str2, '50') == crypt_1):
                                    rez = 0
                                    print("" .join(str2))  # PASSWORD
                                    exit(0)
exit(0)